go/parquet/internal/bmi/bmi_arm64.go

```diff
@@ -14,40 +14,38 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !noasm
 // +build !noasm
 
 package bmi
 
 import (
 	"os"
 	"strings"
-)
-import (
-	"golang.org/x/sys/cpu"
+
+	"github.com/klauspost/cpuid/v2"
 )
 
 func init() {
-    // Added ability to enable extension via environment:
+	// Added ability to enable extension via environment:
 	// ARM_ENABLE_EXT=NEON go test
 	if ext, ok := os.LookupEnv("ARM_ENABLE_EXT"); ok {
 		exts := strings.Split(ext, ",")
 
 		for _, x := range exts {
 			switch x {
 			case "NEON":
-				cpu.ARM64.HasASIMD = true
+				cpuid.CPU.Enable(cpuid.ASIMD)
 			case "AES":
-				cpu.ARM64.HasAES = true
+				cpuid.CPU.Enable(cpuid.AESARM)
 			case "PMULL":
-				cpu.ARM64.HasPMULL = true
+				cpuid.CPU.Enable(cpuid.PMULL)
 			default:
-				cpu.ARM64.HasASIMD = false
-				cpu.ARM64.HasAES = false
-				cpu.ARM64.HasPMULL = false
+				cpuid.CPU.Disable(cpuid.ASIMD, cpuid.AESARM, cpuid.PMULL)
```

Comment:
Hmm, so if an invalid value is passed in `ARM_ENABLE_EXT`, all extensions are silently disabled? It would be more user-friendly to print a warning.

Reply:
technically we aren't actually *disabling* the extensions on the CPU, we're just telling the library that anything that wants to check for those extensions should consider them disabled and act accordingly. For example if an invalid value is passed we behave as if those extensions weren't enabled regardless and default to the pure go implementation.

That said, I agree it would be more user-friendly to have an explicit "DISABLE" value, and then just print a warning for invalid values. I'll make that change.

Reply:
> technically we aren't actually _disabling_ the extensions on the CPU

Yes, I figured that out :-)

Reply:
updated so that if the value of the env var is "DISABLE" then we disable things, otherwise we just print any unrecognized options to stderr